### PR TITLE
Adds delete attachment checkbox and functionality only if bulk delete…

### LINKED ISSUE #205: Ability to delete attachments of posts that are getting deleted
Requested at https://wordpress.org/support/topic/delete-post-attachments-too/

### PR DESCRIPTION
**Pull request by sudar**
… attachments plugin is activated

Fix #205

### CODE REVIEW

File: include/Core/Base/Mixin/Renderer.php [hunk 1]
```diff
@@ -345,6 +345,13 @@ protected function render_filtering_table_header() {
 		bd_render_filtering_table_header();
 	}
 
+	/**
+	 * Render attachment settings.
+	 */
+	protected function render_attachment_settings() {
+		do_action( 'bd_render_attachment_settings', $this->field_slug, $this->item_type );
```
Comment (sudar) [inline include/Core/Base/Mixin/Renderer.php]: Pass the entire module object as a parameter. (ie) Just pass `$this` as a parameter to the action.

Also add PHPDoc block to the action.

File: include/Core/Posts/Modules/DeletePostsByCategoryModule.php [hunk 2]
```diff
@@ -54,6 +54,7 @@ public function render() {
 			<table class="optiontable">
 				<?php
 				$this->render_filtering_table_header();
+				$this->render_attachment_settings();
```
Comment (sudar) [inline include/Core/Posts/Modules/DeletePostsByCategoryModule.php]: Let's show the checkbox after the Move/Delete ratio buttons.

File: include/Core/Posts/PostsModule.php [hunk 3]
```diff
@@ -34,6 +34,7 @@
 	protected function parse_common_filters( $request ) {
 		$options = array();
 
+		$options['attachment']   = bd_array_get_bool( $request, 'smbd_' . $this->field_slug . '_attachment', false );
```
Comment (sudar) [inline include/Core/Posts/PostsModule.php]: Do this using a filter.

File: include/Core/Posts/Modules/DeletePostsByCategoryModule.php (hunk 2 above)
Comment (esudalaimadan) [reply include/Core/Posts/Modules/DeletePostsByCategoryModule.php]: Moved delete attachment checkbox after Move/Delete radio buttons.

File: include/Core/Posts/PostsModule.php (hunk 3 above)
Comment (esudalaimadan) [reply include/Core/Posts/PostsModule.php]: You already fixed this.

File: assets/js/src/bulk-delete.js [hunk 4]
```diff
@@ -73,6 +73,17 @@ jQuery(document).ready(function () {
 		}
 	}
 
+	/**
```
Comment (sudar) [inline assets/js/src/bulk-delete.js]: Let's move this JavaScript change to the add-on.

File: include/Core/Base/Mixin/Renderer.php [hunk 5]
```diff
@@ -371,6 +371,20 @@ protected function render_filtering_table_header() {
...
 	}
 
+	/**
+	 * Render attachment settings.
+	 */
+	protected function render_attachment_settings() {
+		/**
+		 * This action is primarily for adding delete attachment settings.
+		 *
+		 * @since 1.3.0
```
Comment (sudar) [inline include/Core/Base/Mixin/Renderer.php]: This should be v6.0.0

File: include/Core/Posts/PostsModule.php [hunk 6]
```diff
@@ -110,10 +110,37 @@ protected function do_delete( $options ) {
...
-		$post_ids = $this->query( $query );
+		$query          = $this->build_query_options( $options, $query );
+		$post_ids       = $this->query( $query );
+		$attachment_ids = array();
+
+		/**
+		 * Triggered before the posts deletion, to get IDs of attachments associated with
+		 * posts that are going to be deleted.
+		 *
+		 * @since 1.3.0
```
Comment (sudar) [inline include/Core/Posts/PostsModule.php]: This should be v6.0.0

File: include/Core/Posts/PostsModule.php [hunk 7]
```diff
@@ -110,10 +110,37 @@ protected function do_delete( $options ) {
...
+		 * @param array $options        List of Delete Options.
+		 */
+		$attachment_ids = apply_filters( 'bd_before_deleting_posts', $attachment_ids, $post_ids, $options );
+
+		$delete_post_count = $this->delete_posts_by_id( $post_ids, $options['force_delete'] );
+
+		/**
+		 * Triggered after the posts are deleted.
+		 *
+		 * @since 1.3.0
```
Comment (sudar) [inline include/Core/Posts/PostsModule.php]: This should be v6.0.0

File: include/Core/Posts/PostsModule.php [hunk 8]
```diff
@@ -110,10 +110,37 @@ protected function do_delete( $options ) {
...
+		 *
+		 * @param array $attachment_ids List of attachment ids that have to be deleted.
+		 * @param array $options        Delete Options.
+		 */
+		if ( ! empty( $attachment_ids ) ) {
+			do_action( 'bd_after_deleting_posts', $attachment_ids, $options );
```
Comment (sudar) [inline include/Core/Posts/PostsModule.php]: Trigger this always. Even if the attachment ids are empty.

**Comment by sudar** [comment]
@esudalaimadan 

Let's make the following changes.

- Remove the render function and add the action inside the `render_delete_settings` function of the `Renderer` class. Not inside the `bd_render_delete_settings` helper function.
- Remove attachment ids from base plugin and store it using an class instance variable in the add-on

File: include/Core/Base/Mixin/Renderer.php (hunk 5 above)
Comment (esudalaimadan) [reply include/Core/Base/Mixin/Renderer.php]: Fixed.

File: include/Core/Posts/PostsModule.php (hunk 6 above)
Comment (esudalaimadan) [reply include/Core/Posts/PostsModule.php]: Fixed.

File: include/Core/Posts/PostsModule.php (hunk 7 above)
Comment (esudalaimadan) [reply include/Core/Posts/PostsModule.php]: Fixed.

File: include/Core/Posts/PostsModule.php (hunk 8 above)
Comment (esudalaimadan) [reply include/Core/Posts/PostsModule.php]: Fixed.

**Comment by esudalaimadan** [comment]
@sudar ,  I have done all the changes in review comments except JS push to bulk-delete-attachments as it currently does not have assets or js folder.
Also, please review the tooltip text and provide your input.

**Pull request merged**